core/trino-main/src/main/java/io/trino/execution/QueryManagerConfig.java

```diff
@@ -52,10 +55,10 @@
 })
 public class QueryManagerConfig
 {
-    public static final long AVAILABLE_HEAP_MEMORY = Runtime.getRuntime().maxMemory();
+    public static final long AVAILABLE_HEAP_MEMORY = getRuntime().maxMemory();
     public static final int MAX_TASK_RETRY_ATTEMPTS = 126;
     public static final int FAULT_TOLERANT_EXECUTION_MAX_PARTITION_COUNT_LIMIT = 1000;
-
+    public static final int DISPATCHER_THREADPOOL_MAX_SIZE = max(10, getRuntime().availableProcessors() * 10);
```

Comment:
interestingly, for `max(50, getRuntime().availableProcessors() * 10);`

then `test(core/trino-main)` was failing, 

```
java.util.concurrent.RejectedExecutionException: Task io.airlift.concurrent.BoundedExecutor$$Lambda/0x00007fe3eba89d80@2d0c38f7 rejected from java.util.concurrent.ThreadPoolExecutor@4da4a424[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 18]
[223](https://github.com/trinodb/trino/actions/runs/8678761480/job/23796182021?pr=21228#step:5:224)
	at java.base/java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2081)

```

https://github.com/trinodb/trino/actions/runs/8678761480/job/23796182021?pr=21228

tried with `max(10, getRuntime().availableProcessors() * 10);` 
and it was successful.

Reply:
> tried with max(10, getRuntime().availableProcessors() * 10);
and it was successful.

That seems random TBH. Which test was failing? What if you repeat it 100 times? Was there some kind of failure that could put `BoundedExecutor` in failed state?

Reply:
thanks @sopel39 
yeah, it was the `TestQueryManagerConfig.testDefaults` test-case, fixed it.

thanks @dain for pointing out.